python/ray/serve/multiplex.py

```diff
@@ -115,32 +161,53 @@ async def load_model(self, model_id: str) -> Any:
-            # models on the current replica has reached the limit.
-            if (
-                self.max_num_models_per_replica > 0
-                and len(self.models) >= self.max_num_models_per_replica
-            ):
-                # Unload the least recently used model.
-                self.models_unload_counter.inc()
-                unload_start_time = time.time()
-                await self.unload_model()
-                self.model_unload_latency_s.set(time.time() - unload_start_time)
-            # Load the model.
-            logger.info(f"Loading model '{model_id}'.")
-            self.models_load_counter.inc()
-            load_start_time = time.time()
-            if self.self_arg is None:
-                self.models[model_id] = await self._func(model_id)
-            else:
-                self.models[model_id] = await self._func(self.self_arg, model_id)
+            # Set the flag to push the multiplexed replica info to the controller
+            # before loading the model.
             self._push_multiplexed_replica_info = True
-            self.model_load_latency_s.set(time.time() - load_start_time)
-        return self.models[model_id]
+            self._model_load_tasks.add(model_id)
+            async with self._model_cache_lock:
+                # Check if the model has been loaded by another request.
+                if model_id in self.models:
+                    return self.models[model_id]
+                try:
+                    # If the number of models per replica is specified, check
+                    # if the number of models on the current replica has
+                    # reached the limit.
+                    if (
+                        self.max_num_models_per_replica > 0
+                        and len(self.models) >= self.max_num_models_per_replica
+                    ):
+                        # Unload the least recently used model.
+                        await self.unload_model()
+                        self._push_multiplexed_replica_info = True
+
+                    # Load the model.
+                    logger.info(f"Loading model '{model_id}'.")
+                    self.models_load_counter.inc()
+                    load_start_time = time.time()
+                    if self.self_arg is None:
+                        self.models[model_id] = await self._func(model_id)
+                    else:
+                        self.models[model_id] = await self._func(
+                            self.self_arg, model_id
+                        )
+                    self._model_load_tasks.discard(model_id)
+                    self.model_load_latency_s.set(time.time() - load_start_time)
+                    return self.models[model_id]
+                except Exception as e:
+                    logger.error(
+                        f"Failed to load model '{model_id}'. Error: {e}",
+                    )
```

Comment:
you can use `logger.exception` here to auto-format the traceback:
```suggestion
                    logger.exception(
                        f"Failed to load model '{model_id}'.",
                    )
```